### PR TITLE
Use a custom LDAP filter during a DCSync in secretsdump.py

### PR DESCRIPTION
**Pull request by snovvcrash**
Reopening this PR from #1090 due to merging conflicts.

@ShutdownRepo could you check if DCSync still happens twice for you, please?

### CODE REVIEW

**Review by 0xdeaddood — CHANGES_REQUESTED** [review]
Hi @snovvcrash!

Thanks for the PR, the addition is great. I have a couple of change requests before merging. Please check it out and let me know what do you think.

File: impacket/examples/secretsdump.py [hunk 1]
```diff
@@ -626,6 +629,34 @@ def getMembersInAlias(self, rid):
 
         return resp
 
+    def getDomainUsersLDAP(self, searchFilter):
```
[the remark this answers is not in the document]
Comment (snovvcrash) [reply impacket/examples/secretsdump.py:632]: Added the check within the commit below.

**Comment by snovvcrash** [comment]
Hey @0xdeaddood, thanks for the review!

Agree with your comments, I've resolved the changes 👍🏻

**Review by 0xdeaddood — APPROVED** [review]

**Comment by 0xdeaddood** [comment]
Merging 🎉🎈 Thanks @snovvcrash!

**Pull request merged**